homeassistant/components/sonos/__init__.py

```diff
@@ -263,8 +263,10 @@ async def _async_create_discovered_player(self, uid, discovered_ip, boot_seqnum)
             else:
                 async_dispatcher_send(self.hass, f"{SONOS_SEEN}-{uid}")
 
-    @callback
-    def _async_ssdp_discovered_player(self, info):
+    async def _async_ssdp_discovered_player(self, info, change):
+        if change == ssdp.SsdpChange.BYEBYE:
```

Comment:
@jjlawren This might be useful to mark sonos devices unavailable right away if it works.  Not sure how useful it will prove to be, but I wanted to make sure you saw this change.

Reply:
@bdraco have you seen this sent out in practice? I'm trying to think of scenarios where a Sonos speaker would have the chance to mark itself as unavailable in normal conditions. The most useful situation I can think of is a power loss, but I don't see this being sent out in that case. Maybe a reboot after firmware upgrade?

Reply:
I haven't actually seen it happen. I expect firmware upgrade will be the case